odl/trafos/fourier.py

```diff
@@ -1645,6 +1647,211 @@ def inverse(self):
             sign=sign, tmp_r=self._tmp_r, tmp_f=self._tmp_f)
 
 
+class NonUniformFourierTransformBase(Operator):
+    """Non uniform Fast Fourier Transform.
+
+    The normalization is inspired from pysap-mri, mainly this class:
+    https://github.com/CEA-COSMIC/pysap-mri/blob/master/mri/reconstruct/fourier.py#L123
+    """
+    def __init__(self, shape, non_uniform_samples, domain, range):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        shape : tuple
+            The dimensions of the data whose non uniform FFT you want to
+            compute
+        non_uniform_samples : iterable
+            List of the fourier space positions where the coefficients are
+            computed. Each position must be in [-0.5; 0.5[
```

Comment:
Ok makes sense!

I don't understand what you mean by normalised **based on Nyquist**. Is [this](https://github.com/CEA-COSMIC/pysap-mri/blob/master/mri/reconstruct/utils.py#L78-L105) what you mean?

Reply:
Yes that's the idea. Note, however, that there are different definitions of Fourier transform, and they result in different notions of "frequency". We go with the mathematician's definition that makes the transform unitary.
Concretely, for us the bandwidth of a uniformly spaced signal is `pi / s`, where `s` is the grid spacing. [Here](https://odlgroup.github.io/odl/math/trafos/fourier_transform.html) are some details. That means that given frequencies should be interpreted as lying in `[-pi/s, pi/s[` and normalized accordingly. (Note that `s` can be different per axis.)

NB: The term "Nyquist" is not mentioned there, I just threw it in sloppily to summarize that whole complex here :-).

Reply:
The most important thing here is that we make sure that this is equivalent to our regular fourier transform if the points are selected in a uniformly spaced manner

Reply:
Right, up to the additional factor that comes from the type of discretization (nearest/linear/...). IMO that has only lead to confusion in the past and should be removed from the existing FT.

Reply:
So I implemented the normalization [here](https://github.com/odlgroup/odl/pull/1488/files?file-filters%5B%5D=.cfg&file-filters%5B%5D=.py#diff-7533defd16313b3c1ec23dd40014de57R71). I took into account the fact that we could want different bandwidth on each dimension.

However I tested only against `numpy` fft for the uniformly spaced manner, because it seemed weird to me to test againsta `odl` fft. Do you want to have a test confirming consistency?